src/Stratis.Bitcoin.Features.Api/Program.cs

```diff
@@ -9,18 +11,26 @@ namespace Stratis.Bitcoin.Features.Api
 {
     public class Program
     {
-        public static void Main(string[] args)
-        {
-        }
-
-        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode, ApiSettings apiSettings)
+        public static IWebHost Initialize(IEnumerable<ServiceDescriptor> services, FullNode fullNode,
+            ApiSettings apiSettings, ICertificateStore store, IWebHostBuilder webHostBuilder = null)
         {
             Guard.NotNull(fullNode, nameof(fullNode));
 
             Uri apiUri = apiSettings.ApiUri;
 
-            IWebHost host = new WebHostBuilder()
-                .UseKestrel()
+            X509Certificate2 certificate = apiSettings.UseHttps ? GetHttpsCertificate(apiSettings, store) : null;
+
+            webHostBuilder = webHostBuilder ?? new WebHostBuilder();
+
+            webHostBuilder
+                .UseKestrel(options =>
+                    {
+                        if (!apiSettings.UseHttps) return;
+                        options.Listen(
+                            IPAddress.Loopback,
```

Comment:
Is this correct? Only loopback? I would actually expect all the normal interfaces the node is listening on because it actually doesn't make sense on the loopback, it only makes sense on the other interfaces. There are Bind addresses in the configuration – we should listen on those if they are provided. If they are not provided, we should listen on all interfaces

Reply:
Yes correct, that was definitely an issue
Looking in there I have also seen a lot of options that could make sense for us too:
options.Limits.MaxConcurrentConnections, options.Limits.MaxRequestBodySize
Might be worth exploring...